### ISSUE TITLE
SSL timeout for login.microsoft.com and login.microsoftonline.com on OTP 25.0

### ISSUE THREAD

**Issue by jamilbk** [erlang/otp#6065]
**Describe the bug**
SSL timeouts connecting to certain hosts, namely `https://login.microsoftonline.com` and `https://login.microsoft.com`

**To Reproduce**
1. Issue a get request using `hackney` or any other HTTP client to `https://login.microsoftonline.com`

**Expected behavior**
SSL connects successfully

**Affected versions**
25.0

**Additional context**
[Here's a barebones Mix project to highlight the issue](https://github.com/firezone/otp_25_ssl_bug) (sorry, I know it's not pure Erlang, but this issue seems to arise from OTP 25's SSL implementation as this bug is not present on OTP <= 24).

Testing conduction on an M1 mac (OpenSSL 3.0.3) and x64_64 linux host (OpenSSL 1.1.1f) both exhibiting the same issue.

Unsure how widespread the issue is -- preliminary testing only shows the two Microsoft URLs triggering the problem.

**Comment by jamilbk** [comment]
cc @princemaple

**Comment by IngelaAndin** [comment]
I could not reproduce this with OTP-25.0.1

**Comment by princemaple** [comment]
Looks like is indeed fixed in 25.0.1 @jamilbk 

```
Erlang/OTP 25 [erts-13.0.1] [source] [64-bit] [smp:24:24] [ds:24:24:10] [async-threads:1] [jit]

Interactive Elixir (1.13.4) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> Mix.install [:hackney]
Could not find Hex, which is needed to build dependency :hackney
Shall I install Hex? (if running non-interactively, use "mix local.hex --force") [Yn] 
* creating /root/.mix/archives/hex-1.0.1
Resolving Hex dependencies...
Dependency resolution completed:
New:
  certifi 2.9.0
  hackney 1.18.1
  idna 6.1.1
  metrics 1.0.1
  mimerl 1.2.0
  parse_trans 3.3.1
  ssl_verify_fun 1.1.6
  unicode_util_compat 0.7.0
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting parse_trans (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
Could not find "rebar3", which is needed to build dependency :unicode_util_compat
I can install a local copy which is just used by Mix
Shall I install rebar3? (if running non-interactively, use "mix local.rebar --force") [Yn] 
* creating /root/.mix/rebar
* creating /root/.mix/rebar3
===> Analyzing applications...
===> Compiling unicode_util_compat
===> Analyzing applications...
===> Compiling idna
===> Analyzing applications...
===> Compiling mimerl
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Analyzing applications...
===> Compiling certifi
===> Analyzing applications...
===> Compiling parse_trans
===> Analyzing applications...
===> Compiling metrics
===> Analyzing applications...
===> Compiling hackney
:ok
iex(2)> :hackney.get("https://login.microsoft.com")
{:ok, 302,
 [
   {"Cache-Control", "no-store, no-cache"},
   {"Pragma", "no-cache"},
   {"Content-Type", "text/html; charset=utf-8"},
   {"Expires", "-1"},
   {"Location", "https://www.office.com/login#"},
   {"Strict-Transport-Security", "max-age=31536000; includeSubDomains"},
   {"X-Content-Type-Options", "nosniff"},
   {"P3P", "CP=\"DSP CUR OTPi IND OTRi ONL FIN\""},
   {"x-ms-request-id", "a4ef1c70-4abf-4224-8294-e866c3774e00"},
   {"x-ms-ests-server", "2.1.12821.11 - AUSELR2 ProdSlices"},
   {"X-XSS-Protection", "0"},
   {"Set-Cookie",
    "fpc=Apa-t3vhJlpHp-tUVObieWs; expires=Sat, 09-Jul-2022 10:18:40 GMT; path=/; secure; HttpOnly; SameSite=None"},
   {"Set-Cookie",
    "esctx=AQABAAAAAAD--DLA3VO7QrddgJg7WevrrB0r_1JkMcyFBNRN7lMvAnDE_CXng0CvlGLjv1fD7hKfn3hAecwuehr3AejTpMO4u2NQIOxuy-CmhPuP8pqwJ1AVQBIIM9eNj3MDcbK-b3Iil42ygSQkpo-b_RBJpK4iimtFh4iJh_FD1zzUkucUOoIY-Bnju0AKBZf_b4BO_FYgAA; domain=.login.microsoft.com; path=/; secure; HttpOnly; SameSite=None"},
   {"Set-Cookie",
    "x-ms-gateway-slice=estsfd; path=/; secure; samesite=none; httponly"},
   {"Set-Cookie",
    "stsservicecookie=estsfd; path=/; secure; samesite=none; httponly"},
   {"Date", "Thu, 09 Jun 2022 10:18:40 GMT"},
   {"Content-Length", "146"}
 ], #Reference<0.3447965295.3999531009.212788>}
iex(3)> :hackney.get("https://login.microsoftonline.com")
{:ok, 302,
 [
   {"Cache-Control", "no-store, no-cache"},
   {"Pragma", "no-cache"},
   {"Content-Type", "text/html; charset=utf-8"},
   {"Expires", "-1"},
   {"Location", "https://www.office.com/login#"},
   {"Strict-Transport-Security", "max-age=31536000; includeSubDomains"},
   {"X-Content-Type-Options", "nosniff"},
   {"P3P", "CP=\"DSP CUR OTPi IND OTRi ONL FIN\""},
   {"x-ms-request-id", "6007d332-5406-4954-ab5e-c4e20ece4000"},
   {"x-ms-ests-server", "2.1.12851.7 - AUSELR1 ProdSlices"},
   {"X-XSS-Protection", "0"},
   {"Set-Cookie",
    "fpc=AmEm6RHqiCFIo_QIqr-moxE; expires=Sat, 09-Jul-2022 10:18:52 GMT; path=/; secure; HttpOnly; SameSite=None"},
   {"Set-Cookie",
    "esctx=AQABAAAAAAD--DLA3VO7QrddgJg7WevrMyl39IGY7hXDpJ_QrOeo1v_S0NHd6w6j8To9z_w-MHifH80CT7LLpY_EeQm0KzPAm-LFxluVt4t4M0iwlgJJYuqXlXHWHY2fR60VusxvZYFy9A-tND7pN-0zM3OmqIrQC4X1f4lEBKSC4gQwOI4ixzXTKxENal8YPoJZSIIzoPMgAA; domain=.login.microsoftonline.com; path=/; secure; HttpOnly; SameSite=None"},
   {"Set-Cookie",
    "x-ms-gateway-slice=estsfd; path=/; secure; samesite=none; httponly"},
   {"Set-Cookie",
    "stsservicecookie=estsfd; path=/; secure; samesite=none; httponly"},
   {"Date", "Thu, 09 Jun 2022 10:18:52 GMT"},
   {"Content-Length", "146"}
 ], #Reference<0.3447965295.3999531009.212890>}
```

**Comment by zemuldo** [comment]
Had to battle with this for quite some time.
